File: official/landing_pages/carrier_language_api/buy-label.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using EasyPost;
using Newtonsoft.Json;

namespace EasyPostExamples
{
    public class Examples
    {
        public static async Task Main()
        {
            var client = new EasyPost.Client("EASYPOST_API_KEY");

            EasyPost.Models.API.Parcel parcel = await client.Parcel.Create(
              new EasyPost.Parameters.Parcel.Create
              {
                  PredefinedPackage = "Parcel",
                  Weight = 28
              }
            );

            EasyPost.Models.API.Shipment shipment = await client.Shipment.Create(
                new EasyPost.Parameters.Shipment.Create
                {
                    ToAddress = toAddress,
                    FromAddress = fromAddress,
                    Parcel = parcel
                }
            );

            await client.shipment.Buy(shipment.Id, shipment.LowestRate());
        }
    }
}

File: official/landing_pages/carrier_language_api/buy-label.java
package shipments;

import com.easypost.exception.EasyPostException;
import com.easypost.model.Shipment;
import com.easypost.model.Parcel;
import com.easypost.service.EasyPostClient;

import java.util.HashMap;

public class Buy {
    public static void main(String[] args) throws EasyPostException {
        EasyPostClient client = new EasyPostClient(System.getenv("EASYPOST_API_KEY"));
        Parcel parcel = client.parcel.create(new HashMap<string, object>() {
            {
                put("predefined_package", "Parcel");
                put("weight", 28);
            }
        });

        Shipment shipment = client.shipment.create(new HashMap<string, object>() {
            {
                put("to_address", toAddress);
                put("from_address", fromAddress);
                put("parcel", parcel);
            }
        });

        client.shipment.buy(shipment.getId(), shipment.lowestRate());
    }
}
